packages/opencensus-exporter-zpages/test/test-zpages.ts

```diff
@@ -0,0 +1,143 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {types} from '@opencensus/opencensus-core';
+import * as assert from 'assert';
+import * as http from 'http';
+import * as mocha from 'mocha';
+
+import {ZpagesExporter, ZpagesExporterOptions} from '../src/zpages';
+
+/** Default options for zpages tests */
+const options = {
+  port: 8081,
+  spanNames: ['predefined/span1', 'predefined/span2'],
+  startServer: false
+} as ZpagesExporterOptions;
+
+/** Zpages Server URL, just to support tests, shouldn't be changed */
+const zpagesServerUrl = 'http://localhost:' + options.port;
+
+/** Default config for traces tests */
+const defaultConfig: types.TracerConfig = {
+  samplingRate: 0.2
+};
+
+/** Zpages tests */
+describe('Zpages Exporter', () => {
+  /** Should create a ZpagesExporter instance with predefined span names */
+  describe('new ZpagesExporter()', () => {
+    const zpages = new ZpagesExporter(options);
+    it('Should create a ZpagesExporter instance', () => {
+      /**
+       * This test ensures that if there are any exceptions during the
+       * constructor execution the test will fail
+       */
+      assert.ok(zpages instanceof ZpagesExporter);
+    });
+    it('Should create predefined span names in the zpages', () => {
+      for (const name of options.spanNames) {
+        /** Array within all same name spans */
+        const spans = zpages.getTraces(name);
+        /** Check if the first position span has the same name */
+        assert.equal(spans[0].name, name);
+      }
+    });
+  });
+
+  /** Should start a new span and get it with zpages */
+  describe('starting a new span', () => {
+    /** Creating here because tracing is a singleton */
+    const tracing = require('@opencensus/nodejs');
+    const zpages = new ZpagesExporter(options);
+
+    tracing.start(defaultConfig);
+    tracing.registerExporter(zpages);
+
+    tracing.tracer.startRootSpan({name: 'rootSpanTest'}, (rootSpan) => {
+      const span = tracing.tracer.startChildSpan('spanNameTest', 'spanType');
+      span.end();
+      rootSpan.end();
+    });
+    it('Should create span in the zpages', () => {
+      /** Array within all same name spans */
+      let spans = zpages.getTraces('rootSpanTest');
+      /** Check if the first position span has the same name */
+      assert.equal(spans[0].name, 'rootSpanTest');
+
+      /** Array within all same name spans */
+      spans = zpages.getTraces('spanNameTest');
+      /** Check if the first position span has the same name */
+      assert.equal(spans[0].name, 'spanNameTest');
+    });
+  });
+
+  describe('catching a running span', () => {
+    /** Creating here because tracing is a singleton */
+    const tracing = require('@opencensus/nodejs');
+    const zpages = new ZpagesExporter(options);
+
+    tracing.start(defaultConfig);
```

Comment:
Put this in `before` rather than `describe` clause.

Reply:
Done.